### PR TITLE
melange: demonstrate issue with ocamldep + melange-ppx

### PR DESCRIPTION
**Pull request by jchavarri**
Showcases some issue that leads to race conditions due to missing dependencies on Melange large codebases.

I am not sure what the fix should be, but I assume `ocamldep` can't be updated to run Melange ppxs, so maybe Dune should start using `meldep` for Melange libraries?

### CODE REVIEW

File: test/blackbox-tests/test-cases/melange/ocamldep-melange-ppx.t [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  main.cmo :
+  main.cmx :
+
+Change main to remove melange-specific ppx:
+
+  $ cat > main.ml <<EOF
+  > let t = A.t
+  > EOF
+
+Now ocamldep does show dependency on A
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/melange/ocamldep-melange-ppx.t:25]: Maybe you can demonstrate that `ocamldep -ppx` works?

**Comment by rgrinberg** [comment]
The test looks alright but I'm not sure if it should live in the dune repo in its current form. It's not really exercising the rules in any way.

**Comment by jchavarri** [comment]
@rgrinberg thanks for the feedback. i was able to reproduce quite minimally, the trick was to make sure the module name used inside Melange `bs.obj` attribute comes after (alphabetically) the module that depends on it. Please check https://github.com/ocaml/dune/pull/6625/commits/9d8ebca6c4c4e9a59b57bb127078b71f2efa082a.

**Comment by rgrinberg** [comment]
@anmonteiro did we decide that the fix for this was to use meldep or to run the melange ppx and then ocamldep?

**Comment by anmonteiro** [comment]
This was fixed with `melange.ppx`.

**Pull request closed without merging**